system/lib/webgpu/webgpu.cpp

```diff
@@ -9,8 +9,19 @@
 
 #include <webgpu/webgpu.h>
 
+#include <array>
 #include <cassert>
 
+static constexpr std::array<WGPUTextureFormat, 3>
+    kBGRA8UnormPreferredContextFormats = {WGPUTextureFormat_BGRA8Unorm,
+                                          WGPUTextureFormat_RGBA8Unorm,
+                                          WGPUTextureFormat_RGBA16Float};
+
+static constexpr std::array<WGPUTextureFormat, 3>
+    kRGBA8UnormPreferredContextFormats = {WGPUTextureFormat_RGBA8Unorm,
+                                          WGPUTextureFormat_BGRA8Unorm,
+                                          WGPUTextureFormat_RGBA16Float};
```

Comment:
nit: Define these inside `wgpuSurfaceGetCapabilities` since they're only used there. (They're static so you can still return pointers to them.)